docs/docs/mutations.md

```diff
@@ -0,0 +1,433 @@
+storeName.mutationName.value; // or other signals
+mutationName.value; // ^^^
+```
+
+#### Callbacks: `onSuccess` and `onError` (optional)
+
+Callbacks can be used on success or error of the mutation. This allows for side effects, such as patching/setting
+state like a service's signal or a store's property.
+
+```ts
+export const CounterStore = signalStore(
+  // ...
+  withMutations((store) => ({
+    increment: rxMutation({
+      // ...
+      onSuccess: (result) => {
+        console.log('result', result);
+        patchState(store, { counter: result });
+      },
+    }),
+  })),
+);
+
+@Component({...})
+class CounterMutation {
+  // ...
+  private saveToServer = httpMutation<Params, CounterResponse>({
+    // ...
+    onError: (error) => {
+      console.error('Failed to send counter:', error);
+    },
+  });
+}
+```
+
+#### Flattening operators (optional to specify, has default)
+
+```ts
+// Default: concatOp
+// All options: concatOp, exhaustOp, mergeOp, switchOp
+
+(withMutations((store) => ({
+  increment: rxMutation({
+    // ...
+    operator: concatOp, // default if `operator` omitted
+  }),
+})),
+  class SomeComponent {
+    private saveToServer = httpMutation<Params, CounterResponse>({
+      // ...
+      // Passing in a custom option. Need to import like:
+      // `import { switchOp } from '@angular-architects/ngrx-toolkit'`
+      operator: switchOp,
+    });
+  });
+```
+
+#### Methods
+
+A mutation is its own function to be invoked, returning a promise should you want to await one.
+
+```ts
+@Component({...})
+class CounterRxMutation {
+  private increment = rxMutation({...});
+  private store = inject(CounterStore);
+
+  // To await
+  async incrementBy13() {
+    const resultA = await this.increment({ value: 13 });
+    if (resultA.status === 'success') { ... }
+
+    const resultB = await this.store.increment({ value: 13 });
+    if (resultB.status === 'success') { ... }
+  }
+
+  // or not to await, that is the question
+  incrementBy12() {
+    this.increment({ value: 12 });
+
+    this.store.increment({ value: 12 });
+  }
+}
+```
+
+### Choosing between `rxMutation` and `httpMutation`
+
+Though mutations and resources have different intents, the difference between `rxMutation` and `httpMutation` can be seen in a
+similar way as `rxResource` and `httpResource`
+
+For brevity, take `rx` as `rxMutation` and `http` for `httpMutation`
+
+- `rx` to utilize RxJS streams, `http` to make an `HttpClient` request
+  - `rx` could be any valid observable, even if it is not HTTP related.
+  - `http` has to be an HTTP request. The user's API is agnostic of RxJS. _Technically, HttpClient with observables is used under the hood_.
+- Primary property to pass parameters to:
+  - `rx`'s `operation` is a function that defines the mutation logic. It returns an Observable,
+  - `http` takes parts of `HttpClient`'s method signature, or a `request` object which accepts those parts
+
+<!-- TODO - I was wrong on flattening part, re-write -->
```

Comment:
@rainerhahnekamp after our discussion about my misunderstanding, I think you or @manfredsteyer  would be better suited to write this part instead